File: almostUHM.py
from hashlib import sha224, sha256, sha384, sha512
from re import findall
from typing import Callable

# Create a list of Hash Methods
hash_methods: list = [
    sha224,
    sha256,
    sha384,
    sha512
]

# A function to generate a random Number based on a seed
def randomize(seed: int) -> int:
    hashed_value = sha512(str(seed).encode()).hexdigest()
    regex = r".{1,10}"
    hashed_value = findall(regex, hashed_value)
    hashed_value = sum([int(i,16) for i in hashed_value]) 
    return hashed_value % 2147483647

# A function to generate a Hash function based on a seed
def generateProceduralHashFunction(seed: int) -> Callable[[str], str]:
    """
    Usage:
    import almostUHM
    hash_method = almostUHM.generateProceduralHashFunction(1234567890)
    hash = hash_method('Hello World')
    print(hash)
    """
    seed: int = int(seed)
    seed = randomize(seed)
    def proceduralHashFunction(string: str) -> str:
        combined_string = string + str(seed)
        method = hash_methods[seed % (len(hash_methods)- 1)]
        hashed = method(combined_string.encode()).hexdigest()
        return hashed
    return proceduralHashFunction

File: almostUHM.js
import { createHash } from 'crypto';

function generateHashFunction(method) {
    return (string) => {
        let hash = createHash(method).update(string, 'utf-8');
        return hash.digest('hex');
    };
}

const hash_methods = [
    generateHashFunction("sha224"),
    generateHashFunction("sha256"),
    generateHashFunction("sha384"),
    generateHashFunction("sha512"),
];

function randomize(n) {
    const hashed_value = hash_methods[3](n.toString());
    var hashed_value_split = hashed_value.match(/.{1,10}/g).map(e => parseInt(e,16));
    return hashed_value_split.reduce((a, b) => a+b) % 2147483647;
}

function generatePrecidualHashFunction(seed) {
    seed = randomize(seed);
    function prodedualHashFunction(string) {
        const combined_string = string + seed.toString();
        const method = hash_methods[seed % (hash_methods.length - 1)];
        const hashed = method(combined_string);
        return hashed;
    }
    return prodedualHashFunction;
}

var seed = 1234
var string = "Hello World!"
var hash_method = generatePrecidualHashFunction(seed)
var hash = hash_method(string)
console.log(hash)